t/t2204-job-manager-dummysched-unlimited.t

```diff
@@ -117,10 +167,38 @@ test_expect_success HAVE_JQ 'job-manager: annotate job id 4 (RIRSI)' '
         jmgr_check_no_annotations $(cat job5.id)
 '
 
+# compared to above, note that job id #2 retains annotations, it is
+# cached in job-info
+test_expect_success HAVE_JQ 'job-manager: annotate job id 4-5 in job-info (RIRSS)' '
+        jinfo_check_annotation $(cat job1.id) "sched.resource_summary" "\"1core\"" &&
+        jinfo_check_annotation $(cat job2.id) "sched.resource_summary" "\"1core\"" &&
+        jinfo_check_annotation $(cat job3.id) "sched.resource_summary" "\"1core\"" &&
+        test_must_fail jinfo_check_annotation_exists $(cat job3.id) "sched.reason_pending" &&
+        test_must_fail jinfo_check_annotation_exists $(cat job3.id) "sched.jobs_ahead" &&
+        jinfo_check_annotation $(cat job4.id) "sched.reason_pending" "\"no cores\"" &&
+        jinfo_check_annotation $(cat job4.id) "sched.jobs_ahead" "0" &&
+        jinfo_check_no_annotations $(cat job5.id)
+'
+
+# compared to above, note that job id #2 retains annotations, it is
+# cached in job-info
+test_expect_success HAVE_JQ 'job-manager: annotate job id 4-5 in flux jobs (RIRSS)' '
+        fjobs_check_annotation $(cat job1.id) "annotations.sched.resource_summary" "1core" &&
+        fjobs_check_annotation $(cat job2.id) "annotations.sched.resource_summary" "1core" &&
+        fjobs_check_annotation $(cat job3.id) "annotations.sched.resource_summary" "1core" &&
+        test_must_fail fjobs_check_annotation_exists $(cat job3.id) "annotations.sched.reason_pending" &&
+        test_must_fail fjobs_check_annotation_exists $(cat job3.id) "annotations.sched.jobs_ahead" &&
+        fjobs_check_annotation $(cat job4.id) "annotations.sched.reason_pending" "no cores" &&
+        fjobs_check_annotation $(cat job4.id) "annotations.sched.jobs_ahead" "0" &&
+        fjobs_check_no_annotations $(cat job5.id)
+'
+
+# cancel non-running jobs first, to ensure they are not accidentally run when
+# running jobs free resources.
 test_expect_success 'job-manager: cancel all jobs' '
-        flux job cancel $(cat job1.id) &&
+        flux job cancel $(cat job4.id) &&
         flux job cancel $(cat job3.id) &&
-        flux job cancel $(cat job4.id)
+        flux job cancel $(cat job1.id)
```

Comment:
Just a suggestion, I think `flux job cancelall` takes a `--states` argument, so this test could be shortened and made self documenting with something like (untested):
```
        flux job cancelall --states=SCHED -f &&
        flux job cancelall -f
```